### ISSUE TITLE
What does `nan_color` do for `lines`?

### ISSUE THREAD

**Issue by kapple19** [MakieOrg/Makie.jl#4510]
I would like if it joined the non-`NaN`s in the plot.
Is that generally favourable? I don't know of another behaviour it could do.
Right now it seems to do nothing.

- [x] what version of Makie are you running? (`]st -m Makie`)

0.21.14

- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)

![Image](https://github.com/user-attachments/assets/482e9222-60b9-47a4-a3ba-e70682830b19)

![Image](https://github.com/user-attachments/assets/92a30495-004f-4de3-9275-205ed7f5b4aa)

- [x] What platform + GPU are you on?

11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz
Intel(R) UHD Graphics
NVIDIA RTX A2000 Laptop GPU

**Comment by jkrumbiegel** [comment]
It is for colors that are NaN, not positions that are NaN. Although with lines it doesn't make as much sense to use I think 

```julia
lines(1:5, [1, 4, 2, 5, 3], color = [1, 2, NaN, 4, 5], nan_color = :red)
```

![Image](https://github.com/user-attachments/assets/a2fd4d1d-ed01-47de-9854-97dd81c9408f)

**Comment by kapple19** [comment]
It also then behaves differently as the `nan_color` option for other plot types, is that intentional?

E.g. altering [an example in the docs](https://docs.makie.org/dev/reference/plots/heatmap#Three-vectors),

![Image](https://github.com/user-attachments/assets/da1be5ce-169e-4825-b133-33f59b526e45)

**Comment by SimonDanisch** [comment]
What's the difference?

**Comment by kapple19** [comment]
The `nan_color` for `heatmap` applies to when the data value is `NaN`.
The `nan_color` for `lines` applies to when the colour value is `NaN`.

**Comment by kapple19** [comment]
OH duh, yeah you're right.

That means what I'm proposing as a functionality is inconsistent then 😅

> It is for colors that are NaN, not positions that are NaN. Although with lines it doesn't make as much sense to use I think

It makes sense to me? It's just a more convenient version of users `replace`ing the `NaN`s fed to `lines(color = ...)` with the colour they want.

**Comment by jkrumbiegel** [comment]
I mean that line colors are inherently interpolated, because they are given per point. So then you interpolate between some value-related color and the NaN-replacement color. And that just doesn't make sense because the colors in between mean nothing.

**Comment by kapple19** [comment]
Ah, I understand.
For such a case I would guess I'd want a solid colour for lines that meet with NaN colour points.
Which would make sense visually and you can point out to people "all lines of this colour are <something>".

**Comment by jkrumbiegel** [comment]
Right, that actually makes sense. CairoMakie needs to be updated to that behavior then, because GLMakie already does this (it uses value interpolation, then lookup, while CairoMakie can only do color interpolation):

![Image](https://github.com/user-attachments/assets/5da46f5c-d035-4356-96db-e5527b252df5)
